tests/unit/task/test_diagnostic_task.py

```diff
@@ -90,6 +91,18 @@ def test_initialize_cmd(ext_profile, cmd, tmp_path, monkeypatch):
 
     assert task.cmd == cmd
 
+    # test for no executable
+    monkeypatch.setattr(esmvalcore._task, 'which', lambda x: None)
+    if ext_profile[0] != '':
+        with pytest.raises(DiagnosticError) as err_mssg:
+            task = esmvalcore._task.DiagnosticTask(script,
```

Comment:
```suggestion
            esmvalcore._task.DiagnosticTask(script,
```
since you're not using the variable anywhere